src/cpp/include/tasks/classification.h

```
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2020-2025 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <opencv2/opencv.hpp>
+#include <openvino/openvino.hpp>
+
+#include "adapters/inference_adapter.h"
+#include "tasks/classification/resolvers.h"
+#include "tasks/results.h"
+#include "utils/config.h"
+#include "utils/vision_pipeline.h"
+
+class Classification {
+public:
+    std::shared_ptr<InferenceAdapter> adapter;
+    VisionPipeline<ClassificationResult> pipeline;
+
+    Classification(std::shared_ptr<InferenceAdapter> adapter, cv::Size input_shape)
```

Does it make sense to define a constant like `DEFAULT_INPUT_SHAPE`? Or maybe hide this parameter: I'd opt for avoiding exposing model reshaping feature for now. Only a fraction of Geti models support  reshaping, let alone arbitrary public models

---

Yes, good point. The problem is that its model dependent. But I believe its added during serialization step, so I can just retrieve it again. I was just solving a variable scope issue, but I'll figure it out and apply everywhere.